benchmark/run_bench.sh

```diff
@@ -0,0 +1,207 @@
+#!/bin/bash
+
+rs=(1)
+cs=(1)
+w=10
+d=10
+reqs=(1)
+resps=(1)
+rpc_types=(unary)
+
+# idx[0] = idx value for rs
+# idx[1] = idx value for cs
+# idx[2] = idx value for reqs
+# idx[3] = idx value for resps
+# idx[4] = idx value for rpc_types
+idx=(0 0 0 0 0)
+idx_max=(1 1 1 1 1)
+
+inc()
+{
+  for i in $(seq $((${#idx[@]}-1)) -1 0)
+  do
+    idx[${i}]=$((${idx[${i}]}+1))
+    if [ ${idx[${i}]} == ${idx_max[${i}]} ]; then
+      idx[${i}]=0
+    else
+      break
+    fi
+  done
+  return=1
+  # Check to see if we have looped back to the beginning.
+  for v in ${idx[@]}; do
+    if [ ${v} == 0 ]; then
+      return=$((${return}*1))
+    else
+      return=$((${return}*0))
+    fi
+  done
+  if [ ${return} == 1 ]; then
+  rm -Rf ${out_dir}
+    exit 0
+  fi
+}
+
+run(){
+  nr=${rs[${idx[0]}]}
+  nc=${cs[${idx[1]}]}
+  req_sz=${reqs[${idx[2]}]}
+  resp_sz=${resps[${idx[3]}]}
+  r_type=${rpc_types[${idx[4]}]}
+  # Following runs one benchmark
+  base_port=50051
+  delta=0
+  test_name="r_"${nr}"_c_"${nc}"_req_"${req_sz}"_resp_"${resp_sz}"_"${r_type}"_"$(date +%s)
+  echo "================================================================================"
+  echo ${test_name}
+  while :
+  do
+    port=$((${base_port}+${delta}))
+
+    # Launch the server in background
+    ${out_dir}/server --port=${port} --test_name="Server_"${test_name}&
+    server_pid=$(echo $!)
+
+    # Launch the client
+    ${out_dir}/client --port=${port} --d=${d} --w=${w} --r=${nr} --c=${nc} --req=${req_sz} --resp=${resp_sz} --rpc_type=${r_type}  --test_name="client_"${test_name}
+    client_status=$(echo $?)
+
+    kill ${server_pid}
+    wait ${server_pid}
+
+    if [ ${client_status} == 0 ]; then
+      break
+    fi
+
+    delta=$((${delta}+1))
+    if [ ${delta} == 10 ]; then
+      echo "Continuous 10 failed runs. Exiting now."
+      rm -Rf ${out_dir}
+      exit 1
+    fi
+  done
+
+}
+
+while test $# -gt 0; do
```

Comment:
Use `[ $# -gt 0 ]` instead for consistency with code above or `[[ $# > 0 ]]` for efficiency/clarity, but non-sh-compliance.